src/main/scala/scala/sys/process/ProcessWithPid.scala

```diff
@@ -0,0 +1,48 @@
+package scala.sys.process
+
+import java.lang.{Process => JProcess}
+import sbt.{Level, Logger}
+import scala.util.control.NonFatal
+
+case class ProcessWithPid(process: Process, pid: Option[Long])
+
+object ProcessWithPid {
+  private def reflectJProcess(p: Process.SimpleProcess): JProcess = {
+    val field = p.getClass.getDeclaredField("p")
+    field.setAccessible(true)
+    field.get(p).asInstanceOf[JProcess]
+  }
+
+  // Java 9+ has a `Process#pid()` method, but Java 8 and below have a private `Process#pid` field
+  // We first try to reflect on the method and then fall back to reflecting on the field
+  private def reflectJProcessPid(p: JProcess): Long =
+    try {
+      val method = classOf[JProcess].getMethod("pid")
+      method.invoke(p) match {
+        case pid: java.lang.Long => pid
+        case pid => throw new RuntimeException(s"Expected process PID ($pid) to be a Long, but it was a ${pid.getClass.getName}")
+      }
```

Comment:
If you're content with requiring everyone who develops `sbt-revolver` to use Java 9+ to do so, then this could be simplified to just `p.pid`, plus changing the `catch` block to handle `NoSuchMethodError` when users are on Java 8.

Otherwise, compiling this on Java 8 fails with an error

```
/Users/matt/sbt-revolver/src/main/scala/scala/sys/process/ProcessWithPid.scala:21:9: value pid is not a member of Process
      p.pid
        ^
```